packages/rps/src/redux/game/state.ts

```diff
@@ -1,362 +1,235 @@
-
-// States of the form *A are player A only
-// States of the form *B are player B only
-// All other states are both players
-export enum StateName {
-  NoName = 'NO_NAME',
-  Lobby = 'Lobby',
-  CreatingOpenGame = 'CREATING_OPEN_GAME',
-  WaitingRoom = 'WaitingRoom',
-  WaitForGameConfirmationA = 'WAIT_FOR_GAME_CONFIRMATION_A',
-  ConfirmGameB = 'CONFIRM_GAME_B',
-  DeclineGame = 'DECLINE_GAME_B',
-  WaitForFunding = 'WAIT_FOR_FUNDING',
-  PickWeapon = 'PICK_WEAPON',
-  WaitForOpponentToPickWeaponA = 'WAIT_FOR_OPPONENT_TO_PICK_WEAPON_A',
-  WaitForOpponentToPickWeaponB = 'WAIT_FOR_OPPONENT_TO_PICK_WEAPON_B',
-  WaitForRevealB = 'WAIT_FOR_REVEAL_B',
-  WaitForRestingA = 'WAIT_FOR_RESTING_A',
-  PlayAgain = 'PLAY_AGAIN',
-  OpponentResigned = 'OPPONENT_RESIGNED',
-  WaitForResignationAcknowledgement = 'WAIT_FOR_RESIGNATION_ACKNOWLEDGEMENT',
-  GameOver = 'GAME_OVER',
-  WaitForWithdrawal = 'WAIT_FOR_WITHDRAWAL',
-  PickChallengeWeapon = 'PICK_CHALLENGE_WEAPON',
-  ChallengePlayAgain = 'CHALLENGE_PLAY_AGAIN',
-}
+import { ChannelState, Result, Weapon } from '../../core';
 
-export interface NoName {
-  name: StateName.NoName;
-  libraryAddress: string;
-  myAddress: string;
+export interface GameState {
+  localState: LocalState;
+  channelState: ChannelState | null;
 }
 
-interface NoNameParams {
-  myAddress: string;
-  libraryAddress: string;
-  [x: string]: any;
-}
+export type LocalStateWithPlayer =
+  | GameChosen
+  | OpponentJoined
+  | ChooseWeapon
+  | WeaponChosen
+  | WeaponAndSaltChosen
+  | ResultPlayAgain
+  | WaitForRestart
+  | ShuttingDown
+  | GameOver;
 
-export function noName(obj: NoNameParams): NoName {
-  const { myAddress, libraryAddress } = obj;
-  return { name: StateName.NoName, myAddress, libraryAddress };
-}
+export type LocalState = Empty | Lobby | CreatingOpenGame | WaitingRoom | LocalStateWithPlayer;
+export type PlayingState =
+  | GameChosen
+  | OpponentJoined
+  | ChooseWeapon
+  | WeaponChosen
+  | WeaponAndSaltChosen
+  | ResultPlayAgain
+  | WaitForRestart
+  | ShuttingDown
+  | GameOver;
 
-export interface Lobby {
-  name: StateName.Lobby;
-  myName: string;
-  myAddress: string;
-  twitterHandle: string;
-  libraryAddress: string;
-}
-interface LobbyParams {
-  myName: string;
-  twitterHandle: string;
-  myAddress: string;
-  libraryAddress: string;
-  [x: string]: any;
+export interface Empty {
+  type: 'Empty';
 }
 
-export function lobby(obj: LobbyParams): Lobby {
-  const { myName, myAddress, libraryAddress, twitterHandle } = obj;
-  return { name: StateName.Lobby, myName, myAddress, libraryAddress, twitterHandle };
+export interface Lobby {
+  type: 'Lobby';
+  name: string;
+  address: string;
 }
 
 export interface CreatingOpenGame {
-  name: StateName.CreatingOpenGame;
-  myName: string;
-  twitterHandle: string;
-  myAddress: string;
-  libraryAddress: string;
+  type: 'CreatingOpenGame';
+  name: string;
+  address: string;
+  // libraryAddress: string; // TODO
 }
```

Comment:
This state has been reintroduced (was in legacy code, but disappeared during refactor)

Reply:
See c63e033 , 1ecb37e and ca9befc for similar changes.